device/globals/dialog_instance.gd

```diff
@@ -69,6 +69,27 @@ func finish():
 			anim.play("hide")
 	_queue_free()
 
+func check_clamp(dialog_pos):
```

Comment:
Maybe `clamped_position` would be a better name, since it's not a void function?

Reply:
Or maybe `clamp_position` and update it in place.